src/compositor.rs

```diff
@@ -26,6 +26,18 @@ pub enum SurfaceError {
     Protocol(#[from] InvalidId),
 }
 
+/// An error caused by creating a region.
+#[derive(Debug, thiserror::Error)]
+pub enum RegionError {
+    /// The compositor global is not available.
+    #[error("the compositor global is not available")]
+    MissingCompositorGlobal,
```

Comment:
Not an issue with your pull request, maybe we need an error type to represent these two common variants.